byterun/minor_gc.c

```diff
@@ -607,25 +613,21 @@ void caml_empty_minor_heap_domain (struct domain* domain)
     caml_ev_end("minor_gc/ephemerons");
 
     caml_ev_begin("minor_gc/update_minor_tables");
-    for (r = minor_tables->major_ref.base;
-         r < minor_tables->major_ref.ptr; r++) {
+    for (r = minor_tables->major_ref.base, r_new = minor_tables->major_ref_rewrites.base;
+         r < minor_tables->major_ref.ptr;
+         r++, r_new++) {
+      CAMLassert (r_new < minor_tables->major_ref_rewrites.ptr);
       value v = **r;
       if (Is_block (v) && is_in_interval ((value)Hp_val(v), young_ptr, young_end)) {
-        value vnew;
-        header_t hd = Hd_val(v);
-        int offset = 0;
-        if (Tag_hd(hd) == Infix_tag) {
-          offset = Infix_offset_hd(hd);
-          v -= offset;
-        }
-        CAMLassert (Hd_val(v) == 0);
-        vnew = Op_val(v)[0] + offset;
+        value vnew = (value)*r_new;
         CAMLassert (Is_block(vnew) && !Is_minor(vnew));
         CAMLassert (Hd_val(vnew));
+#ifdef DEBUG
+        header_t hd = Hd_val(v);
```

Comment:
ditto (Hd_val isn't allowed when `v` isn't a block).

I think this assertion block could just be deleted! The logic in `oldify_one` already checks this. (This assertion was useful when the infix tag logic was duplicated here, but that's no longer the case with this PR).

Reply:
agree, removed!